src/openapi-generator.ts

```diff
@@ -842,9 +869,32 @@ export class OpenAPIGenerator {
     return zodSchema.isOptional();
   }
 
+  private getDefaultSchema<T>(zodSchema: ZodTypeAny): T | undefined {
+    if (
+      isZodType(zodSchema, 'ZodOptional') ||
+      isZodType(zodSchema, 'ZodNullable')
+    ) {
+      return this.getDefaultSchema(zodSchema.unwrap());
+    }
+
+    if (isZodType(zodSchema, 'ZodEffects')) {
+      return this.getDefaultSchema(zodSchema._def.schema);
+    }
+
+    if (isZodType(zodSchema, 'ZodDefault')) {
+      return (
+        ('defaultValue' in zodSchema._def && zodSchema._def.defaultValue()) ||
```

Comment:
Is this not properly typed on the Zod size that we need the `in` check?

Reply:
Also I assume we want to use `?? undefined` because defaults of `0` would be ignored. We can add a test for that as well

Reply:
Ah good point. it's because of the stupid && bit which could be false. I was trying to be clever. I'll fix that up

Reply:
Ah looks like when I refactored I no longer needed the `in`. Nice one